src/averageAges.js

```diff
@@ -37,7 +43,19 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  let womenList = people.filter(({ sex }) => sex === 'f');
+
+  const womenMotherList = womenList.filter(({ name }) =>
+    people.some(({ mother }) => mother === name));
+
+  if (withChildren) {
+    womenList = womenMotherList;
+  }
+
+  const summOfAge = womenList
+    .reduce((acc, { died, born }) => acc + died - born, 0);
```

Comment:
You're calling a similar function for men and women, let's put it in a separate helper.